libmobilecoin/src/common/boundary.rs

```diff
@@ -78,10 +78,15 @@ fn ffi_boundary_impl<R>(f: impl (FnOnce() -> Result<R, LibMcError>)) -> Result<R
             // it, we know that no harm will come if we panic while trying to
             // process it.
             let panic_error = AssertUnwindSafe(panic_error);
-            catch_unwind(|| Err(LibMcError::Panic(format!("{:?}", panic_error.0))))
-                // If this also panics then we just abort because at this point it's likely
-                // something terrible has gone wrong and the situation is no longer tenable.
-                .unwrap_or_else(|_| abort())
+            // Rust 2021 disjoint closure capture means a dummy let is needed as
+            // panic_error.0 is not unwind safe
+            catch_unwind(|| {
+                let _ = &panic_error;
+                Err(LibMcError::Panic(format!("{:?}", panic_error.0)))
+            })
```

Comment:
IIUC we can just use dereference the wrapped var, since `AssertUnwindSafe`' implements `Deref`
```suggestion
            catch_unwind(|| Err(LibMcError::Panic(format!("{:?}", *panic_error))))
```

Reply:
Good point!